src/main/java/com/yelp/nrtsearch/server/grpc/LuceneServer.java

```diff
@@ -1307,6 +1307,29 @@ public void stopIndex(
       }
     }
 
+    @Override
+    public void syncIndexState(
+        SyncIndexStateRequest request, StreamObserver<DummyResponse> responseObserver) {
+      if (globalState.getConfiguration().getStateConfig().useLegacyStateManagement()) {
+        responseObserver.onError(
+            Status.UNAVAILABLE.withDescription("legacy state not supported").asRuntimeException());
+        return;
+      }
+      try {
+        if (globalState.getConfiguration().getIndexStartConfig().getMode().equals(Mode.REPLICA)) {
+          globalState.getIndexStateManager(request.getIndexName()).load();
```

Comment:
I think this would work for reloading index state. Do we want to support the case where a new index gets added? This would require reloading the global state, and potentially starting/stopping indices.